modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -159,10 +160,16 @@ export default class GeoJsonLayer extends CompositeLayer {
       updateTriggers
     } = this.props;
 
-    const drawPoints = pointFeatures && pointFeatures.length > 0;
-    const drawLines = lineFeatures && lineFeatures.length > 0;
-    const hasPolygonLines = polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
-    const hasPolygon = polygonFeatures && polygonFeatures.length > 0;
+    const hasPoints =
```

Comment:
This code looks too complex/ad-hoc. We shouldn't need a separate set of checks for `subLayerProps`. Some alignment of the API should be done.

What if we move the calls to `getSubLayerProps` out of the functions below and then just inspect the resolved props, rather than add manual override checking in multiple places, could that be a way?

Reply:
The original logic was to not render the sub layer at all, which `getSubLayerProps` cannot control.